modular_citadel/code/game/machinery/wishgranter.dm

```diff
@@ -0,0 +1,107 @@
+/obj/machinery/wish_granter/attack_hand(mob/living/carbon/user)
+	if(charges <= 0)
+		to_chat(user, "The Wish Granter lies silent.")
+		return
+
+	else if(!ishuman(user))
+		to_chat(user, "You feel a dark stirring inside of the Wish Granter, something you want nothing of. Your instincts are better than any man's.")
+		return
+
+	else if (!insisting)
+		to_chat(user, "Your first touch makes the Wish Granter stir, listening to you.  Are you really sure you want to do this?")
+		insisting++
+
+	else
+		if(is_special_character(user))
+			to_chat(user, "You speak.  [pick("I want power","Humanity is corrupt, mankind must be destroyed", "I want to rule the world","I want immortality")].  The Wish Granter answers.")
+			to_chat(user, "Your head pounds for a moment, before your vision clears. The Wish Granter has given you limitless power, and it's all yours!")
+			user.dna.add_mutation(HULK)
+			user.dna.add_mutation(XRAY)
+			user.dna.add_mutation(COLDRES)
+			user.dna.add_mutation(TK)
+			charges--
+			insisting = FALSE
+			to_chat(user, "You have a very great feeling about this!")
```

Comment:
Tbh, thinking about it a little, you could probably make this add a 0.5 click delay multiplier to make getting the wishgranter a pure reward for antags.